### PR TITLE
Migrate to gotest.tools :)

### PR DESCRIPTION
**Pull request by vdemeester**
Related to https://github.com/docker/libnetwork/pull/2213

- Also cleans some vendor.conf unused lines

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like some of the empty dependencies were added in https://github.com/docker/libnetwork/pull/2183

/cc @euanh

**Comment by euanh** [comment]
@thaJeztah I bumped all dependencies which libnetwork and moby had in common to the version used in moby.   I didn't look for dependencies which could be removed from libnetwork.    The new dependencies at the bottom of the file were added because one of the new versions of an existing dependency needed them.

It seems some of the original dependencies can be removed, taking the new ones with them.   Great!

File: datastore/datastore_test.go [hunk 1]
```diff
@@ -68,12 +68,12 @@ func TestKVObjectFlatKey(t *testing.T) {
 func TestAtomicKVObjectFlatKey(t *testing.T) {
 	store := NewTestDataStore()
 	expected := dummyKVObject("1111", true)
-	assert.False(t, expected.Exists())
+	assert.Check(t, !expected.Exists())
 	err := store.PutObjectAtomic(expected)
 	if err != nil {
```
Comment (thaJeztah) [inline datastore/datastore_test.go]: For someone that has some time to spend; these can be changed to use `assert.NilError(t, err)`

**Comment by fcrisciani** [comment]
rebased and fixed the conflict

**Pull request merged**